Sources/PackageGraph/LocalPackageContainer.swift

```diff
@@ -30,27 +30,35 @@ public final class LocalPackageContainer: PackageContainer {
     private let toolsVersionLoader: ToolsVersionLoaderProtocol
     private let currentToolsVersion: ToolsVersion
 
-    /// The file system that shoud be used to load this package.
+    /// The file system that should be used to load this package.
     private let fileSystem: FileSystem
 
     /// cached version of the manifest
     private let manifest = ThreadSafeBox<Manifest>()
 
     private func loadManifest() throws -> Manifest {
         try manifest.memoize() {
+            let path: AbsolutePath
+            switch self.package.kind {
+            case .root(let _path), .fileSystem(let _path):
+                path = _path
+            default:
+                throw InternalError("invalid package type \(package.kind)")
```

Comment:
Could we add the package identity to the message here (or is it perhaps already part of the metadata?).  That would make it much easier to try to figure out errors coming in from the field that have this message.

Reply:
sure. this is an internal error so a programmer error and not user one, so less important IMO